Woof/Woof/ViewModel/SitterProfileViewModel.swift

```diff
@@ -21,29 +22,54 @@ final class SitterProfileViewModel: ObservableObject {
     /// The price per hour for walking charged by the pet sitter.
     @Published var pricePerHour: String = ""
 
+    /// Indicating whether an error has occurred during the network operation.
+    @Published var isErrorOccurred: Bool = false
+
+    /// Detailed error information for the user.
+    @Published var errorMessage: String = ""
+
     /**
      Initializes an instance of the `SitterProfileViewModel` class.
      */
     init() {
         currentSitter = loadSitterFromStorage()
 
         setInitialValues(currentSitter)
+
+        isErrorOccurred = false
     }
 
-    /// Requests model layer to save modified data.
-    func save() {
-        currentSitter.name = name
-        currentSitter.surname = surname
-        currentSitter.phone = phone
-        currentSitter.bio = bio
-        if let newPricePerHour = Double(pricePerHour) {
-            currentSitter.pricePerHour = newPricePerHour
+    /// Requests model layer to upload and save modified data.
+    func save() async {
+        var updatedSitter = currentSitter
+
+        updatedSitter.name = name
+        updatedSitter.surname = surname
+        updatedSitter.phone = phone
+        updatedSitter.bio = bio
+        updatedSitter.pricePerHour = Double(pricePerHour) ?? currentSitter.pricePerHour
+
+        do {
+            try await upload(updatedSitter)
+
+            currentSitter = updatedSitter
+
+            if let data = try? JSONEncoder().encode(currentSitter) {
+                if KeyValueStorage(KeyValueStorage.Name.currentSitter)
+                    .save(data, for: KeyValueStorage.Key.currentSitter) {
+                } else {
+                    await MainActor.run {
+                        isErrorOccurred = true
+                        handleError(.localSaveFailed)
+                    }
+                }
+            }
+        } catch {
+            await MainActor.run {
+                isErrorOccurred = true
+                handleError(.uploadFailed)
+            }
```

Comment:
See suggestion for `handleError`.
Does it make sense?
```suggestion
            await  handleError(.uploadFailed)
```